lib/compare.js

```diff
@@ -61,7 +60,7 @@ const strictEqualArray = (arr1, arr2) => {
 };
 
 const strictEqualObject = (obj1, obj2) => {
-  if (obj1 === null || obj2 === null) return obj1 === obj2;
+  if (obj1 === null || obj2 === null) return false;
```

Comment:
We should either remove those checks or left them as is because returning `false` is incorrect if passed `(null, null)`. (If we remove this then it will just fail/throw on `Object.keys`)

Reply:
In case it was passed `(null, null)` it will return `true` on the beginning of functions `equal` and `strictEqual`

Reply:
I agree with @lundibundi. Even though this function is not exported we might do this in the future or add a new function that will use this one. IMO, if the function is called `strictEqualObject` then it must be ok to pass `(null, null)` there.

Reply:
IMHO, we do not need it until it is used.In any case, we can return it if needed

Reply:
This 'optimization' will not give anything to the code but instead, make the function behave incorrectly on the valid input data.
As I said I think we should either leave it as is (function correcly handles all input) or remove the statement completely signifying that this function doesn't handle `null`'s and it's an error to pass one here.

Reply:
Since we cannot pass `undefined` or `null` to `Object.keys`, we cannot simply remove this statement. Therefore, this PR does not make sense.